src/contexts/WalletContext/services.ts

```diff
@@ -32,66 +35,76 @@ export const checkForCorrectNetwork = async (provider: ethers.providers.JsonRpcP
  */
 export const checkForRiskyWallet = async (walletAddress: string) => {
   const riskAssessment = await getWalletRiskAssessment(walletAddress);
-  if(isWalletRisky(riskAssessment)) {
+  if (isWalletRisky(riskAssessment)) {
     throw new Error('Risky Account Detected');
   }
 };
 
 /**
  * Will make the wallet open a prompt, asking the user to agree to the T&Cs. It will throw an
- * error if the user disagrees / closes the window. The signature is stored in localStorage to
+ * error if the user disagrees / closes the window. The signature is stored in a database to
  * avoid having to sign the terms every time you connect your wallet.
  * @param signer - The ethers signer
  */
-export const checkForTOSSignature = async (signer: ethers.providers.JsonRpcSigner, walletAddress: string) => {
+export const checkForTOSSignature = async (
+  signer: ethers.providers.JsonRpcSigner,
+  walletAddress: string,
+) => {
   const signerAddress = await signer.getAddress();
   const signatureData = await getSignature(signerAddress);
+  const latestTOS = getTOSText();
   let termsAccepted = false;
 
-  if(signatureData?.signature) {
-    const existingTOSSignerAddress = ethers.utils.verifyMessage(getTOSText(), signatureData.signature);
-    if(existingTOSSignerAddress === signerAddress) {
+  if (!!signatureData?.walletAddress && signatureData.walletAddress === signerAddress) {
+    // Wallet matches
+    if (signatureData.message && signatureData.message === latestTOS) {
+      // Signed data also matches (signature was checked before being written to database)
       termsAccepted = true;
     }
   }
 
-  if(!termsAccepted) {
+  if (!termsAccepted) {
     try {
-      const signature = await signer.signMessage(getTOSText());
-      await saveSignature(signerAddress, signature);
-    } catch(e) {
-      throw new Error(' ');
+      // The latest terms of service have not been accepted. Get a signature now.
+      const termsOfService = getTOSText();
```

Comment:
```suggestion
      const termsOfService = latestTOS; 
```

you already have it a bit above from what I can tell. Perhaps the variable termsOfService in this context is redundant even and we can use `latestTOS`?